### ISSUE TITLE
Code generation error when overriding a nullable primitive property with a non-null one

### ISSUE THREAD

**Issue by SpaceBison** [Kotlin/kotlinx.serialization#522]
**Describe the bug**

A CompilationException is thrown when generating a serializer for a class that overrides a nullable property with a non-null property.

**To Reproduce**
```kotlin
interface IFace {
    val nullablePrimitiveProperty: Long?
}

@Serializable
data class DataClass(override val nullablePrimitiveProperty: Long) : IFace
```

**Result**

```
e: java.lang.IllegalStateException: Backend Internal error: Exception during code generation
Cause: Back-end (JVM) Internal error: wrong code generated
org.jetbrains.kotlin.codegen.CompilationException Back-end (JVM) Internal error: Couldn't transform method node:
write$Self (Lcom/example/kotlinxserializationtest/DataClass;Lkotlinx/serialization/CompositeEncoder;Lkotlinx/serialization/SerialDescriptor;)V:
...
```

Digging into the error messages:

```
Caused by: java.lang.RuntimeException: org.jetbrains.org.objectweb.asm.tree.analysis.AnalyzerException: Error at instruction 15: Argument 3: expected J, but found R
```

which I think corresponds to these lines:

```
INVOKEVIRTUAL com/example/kotlinxserializationtest/DataClass.getNullablePrimitiveProperty ()Ljava/lang/
INVOKEINTERFACE kotlinx/serialization/CompositeEncoder.encodeLongElement (Lkotlinx/serialization/SerialDescriptor;IJ)V (itf)
```

I'd guess that `encodeLongElement()` requires a non-null Long, but we're getting a nullable Long? instead.

**Environment**
 - Kotlin version: 1.3.41
 - Library version:  0.11.0
 - Kotlin platforms: JVM
 - Gradle version: 5.1.1

**Comment by pdvrieze** [comment]
I strongly suspect this has to do with the fact that a non-null long value is normally mapped to the primitive in Java. However that would make for invalid inheritance so it will have to return a wrapper that is marked as non-null (so not a primitive).

**Comment by sandwwraith** [comment]
This is most likely be fixed by the same as in #434 , in 1.3.50
